### ISSUE TITLE
Error: "allow_nested_items_to_be_public" and "min_tls_version" are not supported for a Storage Account located in "AzureCloud"

### ISSUE THREAD

**Issue by jkingdon-ms** [hashicorp/terraform-provider-azurerm#23764]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.2

### AzureRM Provider Version

3.78.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
terraform {
  backend "local" {}

  required_version = ">= 1.6.2"
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "= 3.78.0"
    }
  }
}

provider "azurerm" {
  environment     = "public"
}

resource "azurerm_storage_account" "example_name" {
  name                      = "blah"
  resource_group_name       = "rg_name"
  location                  = "eastus"
  account_kind              = "StorageV2"
  account_tier              = "Standard"
  account_replication_type  = "LRS"
  enable_https_traffic_only = true
  min_tls_version           = "TLS1_2"
}
```

### Debug Output/Panic Output

```shell
`Error: "allow_nested_items_to_be_public" and "min_tls_version" are not supported for a Storage Account located in "AzureCloud"`
```

### Expected Behaviour

According to the terraform documentation, public cloud does support the "allow_nested_items_to_be_public" and "min_tls_version" fields. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account 

I expect to try `terraform apply` and the storage account should deploy without issue.

### Actual Behaviour

I try `terraform apply` and get the Error

`Error: "allow_nested_items_to_be_public" and "min_tls_version" are not supported for a Storage Account located in "AzureCloud"`

### Steps to Reproduce

`terraform apply`

### Important Factoids

Running in Azure Public

### References

Documentation that conflicts with the error: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account

**Comment by magodo** [comment]
@jkingdon-ms I can provision with your config above with no problem. From the error message, the environment name somehow is set to "AzureCloud", which typically will raise error during provider initialization. Please provide a full config and the debug log.

**Comment by randomnote1** [comment]
I was able to reproduce this issue as well. This issue is a blocker for those with custom Azure Clouds who don't have internet access.

The following is the configuration I used:

```HCL
locals {
  resource_group_name  = "<resource group name>"
  storage_account_name = "<storage account name>"
}

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.84.0"
    }
  }
}

provider "azurerm" {
  features {}
  metadata_host              = "management.azure.com"
  skip_provider_registration = true
}

data "azurerm_resource_group" "rg" {
  name = local.resource_group_name
}

resource "azurerm_storage_account" "storage" {
  name                     = local.storage_account_name
  resource_group_name      = data.azurerm_resource_group.rg.name
  location                 = data.azurerm_resource_group.rg.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}
```

**Comment by Smgilliam25** [comment]
Same issue, with custom cloud.

**Comment by holmanjd** [comment]
Bug is still in code @jkingdon-ms, we can duplicate error when moving providers from way back (say 3.39.0) and 3.97.1
I have commented out #min_tls_version and yet it still picks up min_tls_version.

resource "azurerm_storage_account" "loganalytics" {
  name                      = "thing"
  resource_group_name       = var.resource_group_name
  location                  = var.location
  account_kind              = "StorageV2"
  account_tier              = "Standard"
  account_replication_type  = "LRS"
  enable_https_traffic_only = true
  #min_tls_version           = "TLS1_2"
  tags = merge(
    var.tags,
    {
      Description = "vnet loganalytics"
    },
  )
}


│ Error: "allow_nested_items_to_be_public" and "min_tls_version" are not supported for a Storage Account located in "AzureCloud"
│
│   with module.spoke-network-t3.module.spoke-network.azurerm_storage_account.loganalytics,
│   on ..\modules\virtual-network\main.tf line 24, in resource "azurerm_storage_account" "loganalytics":
│   24: resource "azurerm_storage_account" "loganalytics" {
│

**Comment by randomnote1** [comment]
This issue was resolved in v3.99.0.
